FHEM/14_SD_WS.pm

```diff
@@ -737,6 +738,32 @@ sub SD_WS_Parse($$)
 	    Log3 $iohash, 4, "$name: SD_WS_WH2_8 $protocol ($SensorTyp) sensor id=$id, Data:".substr($bitData,$vorpre + 24,8)." hum=$hum";
 	    Log3 $iohash, 4, "$name: SD_WS_WH2_9 $protocol ($SensorTyp) sensor id=$id, channel=$channel, temp=$temp, hum=$hum";
 		
+	} elsif ($protocol eq "94")	# Atech wireless weather station
```

Comment:
Kann das auch als Eintrag in decodingSubs  implementiert werden?

Reply:
Das erschien mir umständlicher, da ich eine Variable in mehreren subs brauche.

Reply:
Du meinst wegen change 110 to 1 ?

Reply:
Genau deshalb.

Reply:
Gefällt mir nicht so richtig. Eigentlich braucht es nur einen "optionalen" Aufruf einer anonymen sub in der das geändert wird und dann kann es über einen Eintrag in decodingSubs realisiert werden.


Was meinst Du?

Reply:
Naja, eigentlich finde ich die vielen Subs eigentlich uneffizient, da werden immer wieder Variablen im Speicher hin- und herkopiert, was eigentlich unnötig ist. Wenn ich eine Variable mehrmals benötige, platziere ich sie so, das ich darauf sofort Zugriff habe.

Reply:
Diese If/Else Schlangen machen es leider etwas schlechter handhabbar und können auch manchmal tolle Überraschungen mitbringen.

Das Kopieren der Variablen können wir vermeiden.

Mir war damals halt wichtig, dass der Code gut lesbar ist, was je Sensor passiert.

Reply:
Soll ich es nun noch einmal umschreiben, oder akzeptierst du es so?

Reply:
Wenn Du nichts dagegn hast, würde ich es umschreiben

Reply:
Tu das, ich kann da nur dazu lernen :-)